Makefile

```diff
@@ -173,9 +173,20 @@ manifests: controller-gen
 
 # Generate code (API)
 .PHONY: generate
-generate: controller-gen
+generate: controller-gen adapter/generated/openapi/zz_generated.openapi.go
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
+adapter/generated/openapi/zz_generated.openapi.go: go.mod go.sum
+	@OPENAPI_PATH=`go list -m -f '{{.Dir}}' k8s.io/kube-openapi`; \
+	go run $${OPENAPI_PATH}/cmd/openapi-gen/openapi-gen.go --logtostderr \
```

Comment:
the `OPENAPI_PATH` is not resolved, as we can see in the error `stat /cmd/openapi-gen/openapi-gen.go: no such file or directory` in https://github.com/kedacore/keda/pull/1512/checks?check_run_id=1734447773#step:7:11

I had some problems with GO_PATH in the GH Action before, what was working for me locally behaved differently in GH Action :(

Reply:
I narrowed down the problem to `go mod vendor` we are doing in Makefile and not cleaning it up after. I fixed the path resolution problem even when vendor dir is present, but we might want to stop doing the `go mod vendor` thing (or cleanup after that if really required).

Reply:
Awesome, thanks a lot for that. I do recall that the `go mod vendor` was needed to get correct clientset generation. But the vendor cleanup seems like a good idea. Thanks!